index.js

```diff
@@ -31,10 +35,11 @@ module.exports = function packager (opts, cb) {
     default: return cb(new Error('Unsupported platform. Must be either darwin, linux, or win32'))
   }
 
-  // Ignore this and related modules by default
-  var defaultIgnores = ['/node_modules/electron-prebuilt($|/)', '/node_modules/electron-packager($|/)', '/\.git($|/)']
-  if (opts.ignore && !Array.isArray(opts.ignore)) opts.ignore = [opts.ignore]
-  opts.ignore = (opts.ignore) ? opts.ignore.concat(defaultIgnores) : defaultIgnores
+  // Only include files specified
+  var defaultFiles = ['package.json']
+  opts.files = (defaultFiles + config.files).map(function (item) {
+    path.join(opts.dir, item)
```

Comment:
I'm currently working on that now.

On Sat, Jun 20, 2015 at 12:16 PM, maxogden <notifications@github.com> wrote:

> In index.js
> <https://github.com/maxogden/electron-packager/pull/80#discussion_r32886167>
> :
>
> > @@ -31,10 +35,11 @@ module.exports = function packager (opts, cb) {
> >      default: return cb(new Error('Unsupported platform. Must be either darwin, linux, or win32'))
> >    }
> >
> > -  // Ignore this and related modules by default
> > -  var defaultIgnores = ['/node_modules/electron-prebuilt($|/)', '/node_modules/electron-packager($|/)', '/\.git($|/)']
> > -  if (opts.ignore && !Array.isArray(opts.ignore)) opts.ignore = [opts.ignore]
> > -  opts.ignore = (opts.ignore) ? opts.ignore.concat(defaultIgnores) : defaultIgnores
> > +  // Only include files specified
> > +  var defaultFiles = ['package.json']
> > +  opts.files = (defaultFiles + config.files).map(function (item) {
> > +    path.join(opts.dir, item)
>
> Oh I see, you are reading the files field of the package.json and using
> that as the list of files to include. I like that approach but not all
> package.jsons will have the files field populated, so I think we should
> maybe show the user an error or warning if the packager can't find any
> files to include.
>
> Also is there a way to specify the files using flags like in my comment
> above? From reading the code it looks like it only pulls from package.json
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/maxogden/electron-packager/pull/80/files#r32886167>.
>



-- 
Kurtis Rainbolt-Greene, Hacker
Software Developer
1631 8th St.
New Orleans, LA
70115